src/ktopt_planning_context.cpp

```diff
@@ -175,37 +186,129 @@ void KTOptPlanningContext::setRobotDescription(std::string robot_description)
+                                              std::move(meshcat_viz_params));
+  visualizer_ = &visualizer;
 
   // in the future you can add other LeafSystems here. For now building the
   // diagram
-  auto diagram_ = builder.Build();
+  diagram_ = builder->Build();
   diagram_context_ = diagram_->CreateDefaultContext();
-  plant_context_ = &diagram_->GetMutableSubsystemContext(
-    *plant_,
-    diagram_context_.get());
-  nominal_q_ = plant_->GetPositions(*plant_context_);
+  auto& plant_context = diagram_->GetMutableSubsystemContext(plant, diagram_context_.get());
+
+  nominal_q_ = plant.GetPositions(plant_context);
+
+  auto& vis_context = visualizer_->GetMyContextFromRoot(*diagram_context_);
+  visualizer_->ForcedPublish(vis_context);
 }
 
-VectorXd KTOptPlanningContext::toDrakePositions(
-  const moveit::core::RobotState& state,
-  const Joints& joints)
+void KTOptPlanningContext::transcribePlanningScene(const planning_scene::PlanningScene& planning_scene)
+{
+  RCLCPP_INFO(getLogger(), "trying something ...");
+  try
+  {
+    auto world = planning_scene.getWorld();
+  }
+  catch (const std::exception& e)
+  {
+    RCLCPP_ERROR_STREAM(getLogger(), "caought exception ... " << e.what());
+  }
+  auto& plant =
+    dynamic_cast<MultibodyPlant<double>&>(builder->GetMutableSubsystemByName("plant"));
+  auto& scene_graph =
+    dynamic_cast<SceneGraph<double>&>(builder->GetMutableSubsystemByName("scene_graph"));
+  for (const auto& object : planning_scene.getWorld()->getObjectIds())
+  {
+    const auto& collision_object = planning_scene.getWorld()->getObject(object);
+    if (!collision_object)
+    {
+      RCLCPP_ERROR(getLogger(), "No collision object");
+      return;
+    }
+    if (object != OCTOMAP_NS)
+    {
+      RCLCPP_ERROR(getLogger(), "skipping octomap for now ...");
+      for (const auto& shape : collision_object->shapes_)
+      {
+        RCLCPP_ERROR(getLogger(), "iterating inside collision object's shapes");
+        const auto& pose = collision_object->shape_poses_[0];
+        // # drake::geometry::Box box(shape->dimensions_[0], shape->dimenstions_[1], shape->dimensions_[2]);
+        // # const SourceId box1_source_id = scene_graph.RegisterSource("box1");
+
+        // Creates a box geoemtry and ancors it to the world origin. Better
+        // approach is to create a ground object, anchor that, and then anchor
+        // every non-moving entity to the ground plan
+        // TODO: Create and anchor ground entity
+        Vector3d p(0.3, -0.3, 0.5);
+        const SourceId box_source_id = scene_graph.RegisterSource("box1");
+        // const FrameId box_frame_id = scene_graph.RegisterFrame(box_source_id, GeometryFrame("box1_frame"));
+        const GeometryId box_geom_id = scene_graph.RegisterAnchoredGeometry(
+          box_source_id,
+          std::make_unique<GeometryInstance>(
+            RigidTransformd(p),
+            std::make_unique<Box>(
+              0.15,
+              0.15,
+              0.15),
```

Comment:
couple nitpicks:

* Can you call the vector something more descriptive than `p`?
* Is it worth turning those `0.15` into actual variables?

I know this will likely go away soon anyway, as this is just a test shape?

Reply:
Yes, I intend to replace these with values coming from the planning scene object.